public/app/features/alerting/state/alertDef.ts

```diff
@@ -81,8 +81,14 @@ function createReducerPart(model: any) {
   return new QueryPart(model, def);
 }
 
+// state can also contain the state it was mapped from, ie. "Alerting (NoData)" which means the alert rule is "Alerting" but
+// because the alert rule was configured to use this state instead of "NoData"
+function normalizeAlertState(state: string) {
```

Comment:
Good question, and I honestly have no clue but I suspect that some older implementations might use `no_data` and we want to map it to `nodata` instead.

Reply:
We can't use `mapStateWithReasonToBaseState` here because that one works only for the new Alerting states and these annotations have to work with the legacy alerting states until we can remove that code base entirely.